tests/disas.rs

```diff
@@ -227,3 +247,117 @@ impl Test {
         Ok(isa)
     }
 }
+
+/// Which kind of test is being performed.
+#[derive(Default, Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum TestKind {
+    /// Test the CLIF output, raw from translation.
+    #[default]
+    Clif,
+    /// Compile output to machine code.
+    Compile,
+    /// Test the CLIF output, optimized.
+    Optimize,
+}
+
+/// Assert that `wat` contains the test expectations necessary for `funcs`.
+pub fn run_functions(
+    path: &Path,
+    wat: &str,
+    isa: &dyn TargetIsa,
+    kind: TestKind,
+    funcs: &[Function],
+) -> Result<()> {
+    let mut actual = String::new();
+    for func in funcs {
+        match kind {
+            TestKind::Compile => {
+                let mut ctx = cranelift_codegen::Context::for_function(func.clone());
+                ctx.set_disasm(true);
+                let code = ctx
+                    .compile(isa, &mut Default::default())
+                    .map_err(|e| codegen_error_to_anyhow_error(&e.func, e.inner))?;
+                writeln!(&mut actual, "function {}:", func.name).unwrap();
+                writeln!(&mut actual, "{}", code.vcode.as_ref().unwrap()).unwrap();
+            }
+            TestKind::Optimize => {
+                let mut ctx = cranelift_codegen::Context::for_function(func.clone());
+                ctx.optimize(isa, &mut Default::default())
+                    .map_err(|e| codegen_error_to_anyhow_error(&ctx.func, e))?;
+                writeln!(&mut actual, "{}", ctx.func.display()).unwrap();
+            }
+            TestKind::Clif => {
+                writeln!(&mut actual, "{}", func.display()).unwrap();
+            }
+        }
+    }
+    let actual = actual.trim();
+    log::debug!("=== actual ===\n{actual}");
+
+    // The test's expectation is the final comment.
+    let mut expected_lines: Vec<_> = wat
+        .lines()
+        .rev()
+        .take_while(|l| l.starts_with(";;"))
+        .map(|l| {
+            if l.starts_with(";; ") {
+                &l[3..]
+            } else {
+                &l[2..]
+            }
+        })
+        .collect();
+    expected_lines.reverse();
+    let expected = expected_lines.join("\n");
+    let expected = expected.trim();
+    log::debug!("=== expected ===\n{expected}");
+
+    if actual == expected {
+        return Ok(());
+    }
+
+    if std::env::var("CRANELIFT_TEST_BLESS").unwrap_or_default() == "1" {
```

Comment:
Idle thought: if the test is now a Wasmtime test, should this environment variable be something like `WASMTIME_TEST_BLESS`?

(Smallest of small bikeshed color comments, fine either way, just a thought...)